src/operations/blas1/IndexMaxMin.hpp

```diff
@@ -94,6 +94,7 @@ PORTBLAS_INLINE void IndexMaxMin<is_max, is_step0, lhs_t, rhs_t>::eval(
   using element_t =
       typename ResolveReturnType<op, rhs_t>::type::value_t::value_t;
 
+#ifndef __HIPSYCL__
```

Comment:
Do we need the macro here? I think this implementation is never going to be used with AdaptiveCpp because we have changed the `default_cpu.hpp` file.

Reply:
Yes indeed it's never going there, but for some reason the compiler fails if the `shuffle_down` in that block is exposed unconditionally. Unlike the `sycl::reduce_over_group`, the shuffle_down is not recognized by the compiler and thus the error shows at compile time not runtime.